src/net/connection/virtual_connection.rs

```diff
@@ -1,38 +1,50 @@
-use net::{ExternalAcks, LocalAckRecord, NetworkQuality};
-
-use packet::Packet;
-use sequence_buffer::{SequenceBuffer, CongestionData};
+use net::NetworkConfig;
+use infrastructure::{UnreliableChannel, SequencedChannel, ReliableChannel, Fragmentation, DeliveryMethod, Channel};
+use error::{NetworkResult, NetworkErrorKind};
+use packet::header::StandardHeader;
+use packet::{Packet, PacketTypeId, PacketData};
+use protocol_version::ProtocolVersion;
+use packet::header::HeaderReader;
+use std::io::Cursor;
 use std::fmt;
 use std::net::SocketAddr;
 use std::time::{Duration, Instant};
 
 /// Contains the information about a certain 'virtual connection' over udp.
-/// This stores information about the last sequence number, dropped packages, packages waiting for acknowledgement and acknowledgements gotten from the other side.
+/// This connections also keeps track of network quality, processing packets, buffering data related to connection etc.
 pub struct VirtualConnection {
-    pub seq_num: u16,
-    pub dropped_packets: Vec<Packet>,
-    pub waiting_packets: LocalAckRecord,
-    pub their_acks: ExternalAcks,
+    // client information
     pub last_heard: Instant,
     pub remote_address: SocketAddr,
-    pub quality: NetworkQuality,
-    pub congestion_avoidance_buffer: SequenceBuffer<CongestionData>,
-    pub rtt: f32,
+
+    // fragmentation
```

Comment:
I don't think this comment really adds anything

Reply:
I wouldnt have figured out what type this was 😃